components/name-input.jsx

```diff
@@ -14,13 +18,13 @@ var NameInput = React.createClass({
   },
   validate: function() {
     var valid = true;
-    if (!this.state.values.first) {
+    if (this.state.values.first.trim(' ').length === 0) {
```

Comment:
Also, can probably first check if it's false so we can skip checking trim if we know it's false.

It also ensure we can never call trim on a non string value just in case (although right now we are defaulting).

Something like:

```
if (!this.state.values.first || !this.state.values.first.trim()) {}
```